### PR TITLE
CLOUDP-135171: Fix error message for Delete method

### PR DESCRIPTION
**Pull request by Sugar-pack**
Fix test function usage

### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go [hunk 1]
```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
...
 	}
 	if currentSchedule == nil {
-		r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
-		return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api: %v", currentSchedule)
+		if response != nil {
+			r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. err: %w, response: %v", err, response)
+		} else {
+			r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. err: %w", err)
```
Comment (fabritsius) [inline pkg/controller/atlasdeployment/atlasdeployment_controller.go:273]: Well, I suggest we rewrite this section. At this stage the `error` is likely to be <nil> and it's actually better to log something from the response as long as it isn't empty. Response is an object but I think we can just log only status from it.

Comment (Sugar-pack) [reply pkg/controller/atlasdeployment/atlasdeployment_controller.go]: Done. Thank you

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go [hunk 2]
```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
 	}
 	apiScheduleRes.Policies = []mongodbatlas.Policy{apiPolicy}
 
-	currentSchedule, _, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
+	currentSchedule, response, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
 	if err != nil {
 		r.Log.Debugf("unable to delete current backup policy for project: %v:%v, %v", projectID, cName, err)
+		return fmt.Errorf("unable to delete current backup policy for project: %v:%v, %w", projectID, cName, err)
```
Comment (igor-karpukhin) [inline pkg/controller/atlasdeployment/atlasdeployment_controller.go:265]: code dup nit: print the error message into a string and then use it in both calls

Comment (Sugar-pack) [reply pkg/controller/atlasdeployment/atlasdeployment_controller.go]: Done. Thank you!

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go [hunk 3]
```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
 	}
 	apiScheduleRes.Policies = []mongodbatlas.Policy{apiPolicy}
 
-	currentSchedule, _, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
+	currentSchedule, response, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
 	if err != nil {
-		r.Log.Debugf("unable to delete current backup policy for project: %v:%v, %v", projectID, cName, err)
+		errMessage := "unable to delete current backup policy for project"
+		r.Log.Debugf("%s: %v:%v, %v", errMessage, projectID, cName, err)
+		return fmt.Errorf("%s: %v:%v, %w", errMessage, projectID, cName, err)
```
Comment (fabritsius) [inline pkg/controller/atlasdeployment/atlasdeployment_controller.go:266]:
```suggestion
		return fmt.Errorf("%s: %s:%s, %w", errMessage, projectID, cName, err)
```

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go [hunk 4]
```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
...
 	}
 	if currentSchedule == nil {
 		r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
-		return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api: %v", currentSchedule)
+		if response != nil {
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. response status: %s", response.Status)
```
Comment (fabritsius) [inline pkg/controller/atlasdeployment/atlasdeployment_controller.go:271]: `got an empty response from the api` seems like it should be removed from both here and from of the `else` statement. If the request was made `response` is never empty, if the response is empty, the request was never made, but the `error` was supposed to be returned (meaning we will `return` prior to this line).

So we can assume that response is not `nil` on this line, no need to check.

To be fair I'm not sure this logic is correct as `response` is returned both for success and failures, meaning

**Review by fabritsius — CHANGES_REQUESTED** [review]
Left a couple of suggestions :)

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go (hunk 4 above)
Comment (Sugar-pack) [reply pkg/controller/atlasdeployment/atlasdeployment_controller.go]: Done

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go (hunk 3 above)
Comment (Sugar-pack) [reply pkg/controller/atlasdeployment/atlasdeployment_controller.go]: Done

**Review by fabritsius — APPROVED** [review]
Great Fix 👍

**Pull request merged**